ocaml/xapi/xapi_http.ml

```diff
@@ -130,8 +130,16 @@ let assert_credentials_ok realm ?(http_action = realm) ?(fn = Rbac.nofn)
     | None, Some pool_secret, _
       when SecretString.equal pool_secret !Xapi_globs.pool_secret ->
         fn ()
-    | None, Some _, _ ->
-        raise (Http.Unauthorised realm)
+    | None, Some pool_secret, _ -> (
+      (* if a pool secret rotation is in progress, we accept multiple pool secrets *)
+      match !Xapi_globs.psr_pool_secrets with
```

Comment:
Rather than having both `Xapi_globs.pool_secret` and `Xapi_globs.psr_pool_secrets` for almost the same purpose, could we change the former to `Xapi_globs.pool_secrets` and make it a list of all currently accepted secrets?